### ISSUE TITLE
Detect inconsistent overload with TypeVar

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pylance-release#1778]
<h3> Environment data </h3>

-   Language Server version: 2021.9.0
-   OS and version: darwin x64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.10.0rc1
-   python.analysis.indexing: undefined
-   python.analysis.typeCheckingMode: basic

<h3> Code Snippet </h3>

```python
from typing import Literal, TypeVar, overload

T = TypeVar("T", int, str)

@overload
def func(option: Literal["a"], var: str) -> str:
    ...

@overload
def func(option: Literal["b"], var: int) -> str:  # Return type must be: int
    ...

def func(option: Literal["a", "b"], var: T) -> T:
    pass
```

<h3> Expected behaviour </h3>

Overload **2** is inconsistent with function signature.

**Mypy**
```
error: Overloaded function implementation cannot satisfy signature 2 due to inconsistencies in how they use type variables
```

<h3> Actual behaviour </h3>

_no error_

<h3> Additional information </h3>

After changing the return type of oveload **2** to `list`, pylance will emit an error:
```
Overloaded function implementation is not consistent with signature of overload 2
  Function return type "list[Unknown]" is incompatible with type "T@func"
    Type "list[Unknown]" cannot be assigned to type "int* | str*"
      "list[Unknown]" is incompatible with "int*"
      "list[Unknown]" is incompatible with "str*"
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.9.1, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202191-8-september-2021
